examples/client/pkg/lifecycle/lifecycle.go

```diff
@@ -39,7 +39,7 @@ func (o *federatedServiceObserver) OnDelete(fs *types.FederatedService) error {
 // Start starts the client lifecycle.
 func Start(rootCACert string, peerCert string, peerKey string, serverAddr string, insecureSkipVerify bool) {
 	// Prepare the client instance.
-	tlsConfig := tls.PrepareClientConfig(rootCACert, peerCert, peerKey, insecureSkipVerify)
+	tlsConfig := tls.PrepareMutualTLSClientConfig(rootCACert, peerCert, peerKey, insecureSkipVerify)
```

Comment:
@andrewsykim since the new helper functions are not documented anywhere (apart from the issue, that is going to fade away with time), it may be good if at least you refer here that it exists an alternative helper function. Otherwise, I fear nobody will notice.